### PR TITLE
Zedwallet reset flag

### PR DESCRIPTION
**Pull request by mrrovot**
Added a an option to reset when starting the wallet with the flag `./zedwallet -w walletName -p password --reset <block height>`

### CODE REVIEW

**Comment by brandonlehmann** [comment]
Changed base/target branch :)

File: include/ZedwalletTypes.h [hunk 1]
```diff
@@ -65,6 +68,9 @@ struct Config
 
     /* The wallet password */
     std::string walletPass = "";
+
+    /* The reset startHeight */
+    int resetFromHeight = 0;
```
Comment (zpalmtree) [inline include/ZedwalletTypes.h:73]:
```suggestion
    uint64_t resetFromHeight = 0;
```

File: src/zedwallet++/ParseArguments.cpp [hunk 2]
```diff
@@ -93,6 +98,13 @@ ZedConfig parseArguments(int argc, char **argv)
 
         /* We could check if the string is empty, but an empty password is valid */
         config.passGiven = result.count("password") != 0;
+
+        /* check if reset flag was supplied */
+        config.resetGiven = result.count("reset") != 0;
+
+        if (config.resetGiven) {
```
Comment (zpalmtree) [inline src/zedwallet++/ParseArguments.cpp:105]:
```suggestion
        if (config.resetGiven)
        {
```

File: src/zedwallet++/ParseArguments.h [hunk 3]
```diff
@@ -29,6 +32,9 @@ struct ZedConfig
     /* The wallet password */
     std::string walletPass;
 
+    /* The reset block height */
+    uint resetFromHeight;
```
Comment (zpalmtree) [inline src/zedwallet++/ParseArguments.h:36]:
```suggestion
    uint64_t resetFromHeight;
```

File: src/zedwallet++/ParseArguments.cpp [hunk 4]
```diff
@@ -93,6 +98,13 @@ ZedConfig parseArguments(int argc, char **argv)
...
+          config.resetFromHeight = reset;
```
Comment (zpalmtree) [inline src/zedwallet++/ParseArguments.cpp:106]:
```suggestion
            config.resetFromHeight = reset;
```

**Comment by brandonlehmann** [comment]
That’s last CI build I just skipped was for changing a single letter in a comment.

**Pull request merged**